src/backend/marsha/core/models/video.py

```diff
@@ -60,6 +68,24 @@ def get_source_s3_key(self, stamp=None):
         stamp = stamp or to_timestamp(self.uploaded_on)
         return "{pk!s}/video/{pk!s}/{stamp:s}".format(pk=self.pk, stamp=stamp)
 
+    def update_upload_state(self, upload_state, uploaded_on, **extra_parameters):
+        """Manage upload state.
+
+        Parameters
+        ----------
+        upload_state: Type[string]
+            state of the upload in AWS.
+
+        uploaded_on: Type[DateTime]
+            datetime at which the active version of the file was uploaded.
+
+        extra_paramters: Type[Dict]
+            Dictionnary containing arbitrary data sent from AWS lambda.
+        """
+        super().update_upload_state(upload_state, uploaded_on, **extra_parameters)
+        if "resolutions" in extra_parameters:
+            self.resolutions = extra_parameters.get("resolutions")
```

Comment:
Since you have initialized all resolutions in the migration, it would be safer to refuse updates that don't include resolutions!

Reply:
It's not possible because resolutions are updated only when the encode is complete not before. The `update-state` is called after the upload (without resolutions info) and after `lambda-complete`

Reply:
yes but the backend code doesn't know about that... it's safer that the backend code takes care of its own consistency?

Reply:
If we don't do this check the `lambda-encode` call can erase this field and this definitely don't what we want ! We should only update it when `resolutions` is in the api call.

Reply:
ok so it's worse than I thought :-) Is there a way to update resolutions only for an update state with COMPLETE?

Reply:
But this is what we do. The endpoint `update-state` is called by `lambda-encode` with resolutions only if the status is `COMPLETE`. So here we must check is `resolutions` is in the `extra_paremeters` to not set the `resolutions` field to `null`.

Reply:
We do the same for `uploaded_on`